src/zenml/integrations/aws/secrets_stores/aws_secrets_store.py

```diff
@@ -0,0 +1,1047 @@
+
+        Args:
+            secret_name: The name of the secret.
+            scope: The scope of the secret.
+            workspace: The ID of the workspace to which the secret belongs.
+            user: The ID of the user to which the secret belongs.
+            exclude_secret_id: The ID of a secret to exclude from the check
+                (used e.g. during an update to exclude the existing secret).
+
+        Returns:
+            True if a secret with the given name already exists in the given
+            scope, False otherwise, and an error message.
+        """
+        filter = SecretFilterModel(
+            name=secret_name,
+            scope=scope,
+            page=1,
+            size=2,  # We only need to know if there is more than one secret
+        )
+
+        if scope in [SecretScope.WORKSPACE, SecretScope.USER]:
+            filter.workspace_id = workspace
+        if scope == SecretScope.USER:
+            filter.user_id = user
+
+        existing_secrets = self.list_secrets(secret_filter_model=filter).items
+        if exclude_secret_id is not None:
+            existing_secrets = [
+                s for s in existing_secrets if s.id != exclude_secret_id
+            ]
+
+        if existing_secrets:
+
+            existing_secret_model = existing_secrets[0]
+
+            msg = (
+                f"Found an existing {scope.value} scoped secret with the "
+                f"same '{secret_name}' name"
+            )
+            if scope in [SecretScope.WORKSPACE, SecretScope.USER]:
+                msg += (
+                    f" in the same '{existing_secret_model.workspace.name}' "
+                    f"workspace"
+                )
+            if scope == SecretScope.USER:
+                assert existing_secret_model.user
+                msg += (
+                    f" for the same '{existing_secret_model.user.name}' user"
+                )
+
+            return True, msg
+
+        return False, ""
+
+    @track(AnalyticsEvent.CREATED_SECRET)
+    def create_secret(self, secret: SecretRequestModel) -> SecretResponseModel:
+        """Creates a new secret.
+
+        The new secret is also validated against the scoping rules enforced in
+        the secrets store:
+
+          - only one workspace-scoped secret with the given name can exist
+            in the target workspace.
+          - only one user-scoped secret with the given name can exist in the
+            target workspace for the target user.
+
+        Args:
+            secret: The secret to create.
+
+        Returns:
+            The newly created secret.
+
+        Raises:
+            EntityExistsError: If a secret with the same name already exists
+                in the same scope.
+            RuntimeError: If the AWS Secrets Manager API returns an unexpected
+                error.
+        """
+        self._validate_aws_secret_name(secret.name)
+        user, workspace = self._validate_user_and_workspace(
```

Comment:
I know that we usually also verify that this user is the authenticated user (-> you can't create a secret for a user other than yourself). This is already covered by the endpoint logic and we don't really care about the local case?

Reply:
Yes, it's already covered.